docs/guides/features/command.md

```diff
@@ -164,28 +164,61 @@ It also gives you full control over the way your error type gets serialized. In
 
 ## Async Commands
 
+Asynchronous functions are benefical in Tauri to perform heavy work in a manner that doesn't result in UI freezes or slowdowns.
+
 :::note
 
 Async commands are executed on a separate thread using [`async_runtime::spawn`].
 Commands without the _async_ keyword are executed on the main thread unless defined with _#[tauri::command(async)]_.
 
 :::
 
-If your command needs to run asynchronously, simply declare it as `async`:
+
+:::note
+
+You need to be careful when creating asynchronous functions using Tauri. Currently, you cannot include types that are borrowed, such as those with the & symbol, in the constructor of an asynchronous function. Some common examples of types like this are `&str` and `State<'_, Data>`.
+
+:::
+
+
+If your command needs to run asynchronously, simply declare it as `async`, and resolve any issues with borrowed types.
+
+When working with borrowed types, there are two main options.
+
+**Option 1**: Convert the type, such as `&str` to a similar type that is not borrowed, such as `String`. This won't work for all types.
+
+*Example:*
+
+```rust
+// Declare the constructor using String instead of &str, as &str is borrowed and thus unsupported
+#[tauri::command]
+async fn my_custom_command(value: String) {
+  // Call another async function and wait for it to finish
+  let result = some_async_function(value).await;
+  println!("Result: {}", result);
+}
+```
+
+**Option 2**: Wrap the return type in a Result. This one is a bit harder to implement, but works for all types.
+
+If you have no return type, like in our function, use the return type `Result<(), ()>`. Otherwise, use the return type `Result<bool, ()>`, replacing `bool` with whatever type you wish to return.
+
+*Example:*
 
 ```rust
+// Return a Result<(), ()> to bypass the borrowing issue
 #[tauri::command]
-async fn my_custom_command() {
+async fn my_custom_command(value: &str) -> Result<(), ()> {
   // Call another async function and wait for it to finish
-  let result = some_async_function().await;
+  let result = some_async_function(value).await;
   println!("Result: {}", result);
 }
 ```
 
-Since invoking the command from JS already returns a promise, it works just like any other command:
+**Invoking from JS:** Since invoking the command from JS already returns a promise, it works just like any other command:
 
 ```js
-invoke('my_custom_command').then(() => console.log('Completed!'))
+invoke('my_custom_command', {value: "Hello, Async!"}).then(() => console.log('Completed!'))
```

Comment:
```suggestion
invoke('my_custom_command', { value: "Hello, Async!" }).then(() => console.log('Completed!'))
```
I know that's really picky but i have to 😂

Reply:
It does look nicer...